pontoon/base/placeables.py

```diff
@@ -0,0 +1,64 @@
+import re
+
+
+class MozillaPrintfString:
+    parser = re.compile(r"%(?:[1-9]\$)?[S@]")
+
+
+class PythonPrintfString:
+    parser = re.compile(
+        r"""
+            %
+            (?:\(.*?\))?  # mapping key
+            [\#0\-\ +]*   # conversion flags
+            [\d*]*        # minimum field width
+            (?:\.[\d*])?  # precision
+            [hlL]?        # length modifier
+            [acdEeFfGgiorsuXx%]  # conversion type
+        """,
+        re.X,
+    )
+
+
+class PythonFormatString:
+    parser = re.compile(r"{{|}}|{[\w!.,[\]%:$<>+\-= ]*}")
+
+
+class FluentTerm:
+    parser = re.compile(r"({ *-[^}]*})")
+
+
+class FluentFunction:
+    parser = re.compile(r"({ *[A-W0-9\-_]+[^}]*})")
```

Comment:
```suggestion
class FluentPlaceable:
    parser = re.compile(r"{ *[A-Z0-9\-_][^}]*}")
```

Terms are also captured by the function regexp, so let's fix the regexp's remaining bugs and merge them.